src/CommandingOfficers/TrilogyCommander.java

```diff
@@ -0,0 +1,86 @@
+package CommandingOfficers;
+
+import Engine.GameScenario;
+import Engine.UnitMods.UnitDamageModifier;
+import Engine.UnitMods.UnitDefenseModifier;
+import Engine.UnitMods.UnitModifier;
+import Terrain.MapMaster;
+import Units.Unit;
+import Units.UnitDelta;
+import java.util.ArrayList;
+import CommandingOfficers.CommanderInfo.InfoPage;
+
+/** Base class for AW2/3 COs (AW1 is special) */
+public abstract class TrilogyCommander extends Commander
+{
+  private static final long serialVersionUID = 1L;
+  public static final InfoPage TRILOGY_MECHANICS_BLURB = new InfoPage(
+      "CO powers never charge while a power is active.\n"
+    + "On each power activation, each star in your meter costs 1/5 more of the base star cost.\n"
+    + "This continues for 9 activations, after which the star cost settles at double the base star cost.\n"
+    + "CO power activation deducts the *new* power cost rather than the old one.\n"
+    );
+
+  public TrilogyCommander(CommanderInfo info, GameScenario.GameRules rules)
+  {
+    super(info, rules);
+  }
+
+  @Override
+  public int calculateCombatCharge(UnitDelta minion, UnitDelta enemy, boolean isCounter)
+  {
+    if( null != getActiveAbility() )
+      return 0;
+    if( minion == null || enemy == null )
+      return 0;
+
+    int guiHPLoss  = minion.getHealthDamage() / 10;
+    int guiHPDealt =  enemy.getHealthDamage() / 10;
+
+    int power = 0; // value in funds of the charge we're getting
+
+    // Add up the funds value of the damage done to both participants.
+    power += guiHPLoss * minion.unit.getCost() / 10;
+    // The damage we deal is worth half as much as the damage we take, to help powers be a comeback mechanic.
+    power += guiHPDealt * enemy.unit.getCost() / 10 / 2;
+
+    return power;
+  }
+  @Override
+  public int calculateMassDamageCharge(Unit minion, int lostHP)
+  {
+    return 0;
+  }
+
+  protected abstract static class TrilogyAbility extends CommanderAbility
```

Comment:
Since this is referenced/extended by other classes, part of me thinks it should live in its own file.

Then again, all the one that do inherit the definition, so maybe not so urgent.

Might not hurt to eventually rearrange the directory in general though. We have a bunch of COs together with abstract base classes, COAbility, and some other utility stuff. Maybe you already have a plan though, since you are creating some new directories. Got a long term vision here already?

Reply:
Right, and it's `protected` to enforce that.

That's a good idea. No, I don't currently have a long-term plan. (And if I did, it's slightly schizo since Andy is packaged by game.faction and Brenner is packaged only by faction)

I do kind of like packaging by game.faction.
That would imply:
Ave/Strong/etc -> CommandingOfficers.DefendPeace
RuinedCommander -> COs.AW4
Brenner -> COs.AW4.BrennerWolves

Then I guess the rest are generic extension points/etc used by the rest, so they can stay?